server.js

```diff
@@ -11,8 +11,8 @@ app.use(express.urlencoded({ extended: true }));
 app.use(express.json());
 app.use(morgan('dev'));
 
-app.set('base', '/api/v1');
-app.use('/api/v1', routes);
+app.set('base', `${process.env.BASE_URL}`);
+app.use(`${process.env.BASE_URL}`, routes);
```

Comment:
```
app.use(`${process.env.BASE_URL_V1}`, routes);
app.use(`${process.env.BASE_URL_V2}`, v2routes);
app.use(`${process.env.BASE_URL_V3}`, v3routes);
app.use(`${process.env.BASE_URL}`, v4routes);
```

I think this would work